### PR TITLE
PMP: Add function 'non_manifold_vertices'

### PR DESCRIPTION
**Pull request by MaelRL**
Add a new function to collect all non manifold vertices at once. Useful because `is_non_manifold_vertex` runs in linear time.

## Release Management

* Affected package(s): `PMP`
* Feature/Small Feature (if any): [here](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Add_non_manifold_vertices_collector) -- pre-approved date: 2019-07-02 (accepted 2019-07-16)

### CODE REVIEW

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h [hunk 1]
```diff
@@ -1739,6 +1739,71 @@ bool is_non_manifold_vertex(typename boost::graph_traits<PolygonMesh>::vertex_de
...
+
+  // Non-manifoldness can appear either:
+  // - if 'pm' is pinched at a vertex. While traversing the incoming halfedges at this vertex,
+  //   we will meet strictly more than one border halfedge.
+  // - if there are multiple umbrellas around a vertex. In that case, we will find a non-visited
+  //   halfedge that has for target a vertex that is already visited.
+
+  std::set<vertex_descriptor> non_manifold_vertices;
+  boost::unordered_set<vertex_descriptor> vertices_visited;
+  boost::unordered_set<halfedge_descriptor> halfedges_handled;
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h]: what about dynamic properties :)

Comment (MaelRL) [reply Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h]: Right. Probably can update to dynamic pmaps at other places too.

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h [hunk 2]
```diff
@@ -2179,59 +2179,41 @@ std::size_t make_umbrella_manifold(typename boost::graph_traits<PolygonMesh>::ha
...
-/// duplicates all the non-manifold vertices of the input mesh.
+/// collects the non-manifold vertices (if any) present in the mesh. A non-manifold vertex `v` is returned
+/// via one incident halfedge `h` such that `target(h, pm) = v` for all the umbrellas that `v` apppears in
+/// (an <i>umbrella</i> being the set of faces incident to all the halfedges reachable by walking around `v`
+/// using `hnext = prev(opposite(h, pm), pm)`, starting from `h`).
 ///
-/// @tparam PolygonMesh a model of `HalfedgeListGraph` and `MutableHalfedgeGraph`
-/// @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+/// @tparam PolygonMesh a model of `HalfedgeListGraph`
+/// @tparam OutputIterator a model of `OutputIterator` with value type `boost::graph_traits<PolygonMesh>::halfedge_descriptor`
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h]: the value type of an output iterator is void. You can say "accepting elements of type" "or `hedge_descriptor` can be put inside "

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h [hunk 3]
```diff
@@ -2179,59 +2179,41 @@ std::size_t make_umbrella_manifold(typename boost::graph_traits<PolygonMesh>::ha
...
-/// duplicates all the non-manifold vertices of the input mesh.
+/// collects the non-manifold vertices (if any) present in the mesh. A non-manifold vertex `v` is returned
+/// via one incident halfedge `h` such that `target(h, pm) = v` for all the umbrellas that `v` apppears in
+/// (an <i>umbrella</i> being the set of faces incident to all the halfedges reachable by walking around `v`
+/// using `hnext = prev(opposite(h, pm), pm)`, starting from `h`).
 ///
-/// @tparam PolygonMesh a model of `HalfedgeListGraph` and `MutableHalfedgeGraph`
-/// @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+/// @tparam PolygonMesh a model of `HalfedgeListGraph`
+/// @tparam OutputIterator a model of `OutputIterator` holding objects of type `boost::graph_traits<PolygonMesh>::halfedge_descriptor`
```
Comment (maxGimeno) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h]: I think this causes warnings in the doc.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.0-Ic-87.shtml

**Comment by maxGimeno** [comment]
I'd say it's time to update the CHANGES.md file now

**Pull request merged**